### DISCUSSION TITLE
Trigger bucket index update (a.k.a. the compactor) manually?

### DISCUSSION THREAD

**Discussion by padyx** [grafana/mimir#6028] [Help and support] [answered]
Hey all!

We've recently observed a case when the block storage was unavailable for a while in a system running Mimir.

Mimir behaved nicely to all alert rules that evaluated using recent data - they continued to work.
Alert rules which needed older data failed - this was mostly expected.

But when the persistent storage was back online, the rule evaluation continued failing for a while with "Bucket index too old" errors. This eventually resolved when the compactor ran again and updated the bucket index.

But we're considering if there's a way to fix this case faster: Is there a suggested way to trigger the compactor manually without waiting for the `compaction_interval` period? (Apart from redeploying the compactor with a shorter interval configuration)

I checked the runbook here for the [alert](https://grafana.com/docs/mimir/latest/operators-guide/mimir-runbooks/#err-mimir-bucket-index-too-old), but didn't find anything.

**Comment by dimitarvdimitrov** [comment, marked as answer]
what controls the bucket index update is the `-compactor.cleanup-interval`. Another way to trigger this manually is to restart the compactor. It will do a cleanup and update the bucket index upon startup. At this point there is no HTTP API to achieve the same.

**Reply by padyx** [reply]
Great, thank you for the quick reply! ❤️
